packages/react/src/pages/planning-calendar/planning-calendar.tsx

```diff
@@ -0,0 +1,182 @@
+import './planning-calendar.scss';
+
+import React, { useState, useCallback, useMemo } from 'react';
+import { defaultCalendarListItems } from 'dx-template-gallery-data';
+
+import Button from 'devextreme-react/button';
+import Calendar from 'devextreme-react/calendar';
+import Scheduler, { Resource } from 'devextreme-react/scheduler';
+import SpeedDialAction from 'devextreme-react/speed-dial-action';
+import Tooltip from 'devextreme-react/tooltip';
+
+import { ViewType } from 'devextreme/ui/scheduler';
+import { useScreenSize } from '../../utils/media-query';
+
+import { useSchedulerLogic } from './use-scheduler-logic';
+
+import { CalendarList } from '../../components/calendar-list/calendar-list';
+import { LeftSidePanel } from '../../components/side-panel/left-side-panel';
+import { RightSidePanel } from '../../components/side-panel/right-side-panel';
+import { Agenda } from '../../components/agenda/agenda';
+import { TooltipContentTemplate } from '../../components/scheduler-tooltip/scheduler-tooltip';
+
+const views: ViewType[] = ['day', 'workWeek', 'month', 'agenda'];
+interface CalendarListItem {
+  id: number,
+  text: string,
+  color: string,
+  checkboxColor: string,
+}
+
+const onAppointmentFormOpening = (e) => {
+  const editor = e.form.getEditor('calendarId');
+  if (e.appointmentData.calendarId === undefined) {
+    editor.option('value', 0);
+  }
+};
+
+export const PlanningCalendar = () => {
+  const { isXSmall, isSmall } = useScreenSize();
+  const [calendarListItems] = useState(defaultCalendarListItems);
+  const {
+    agendaItems,
+    currentView,
+    date,
+    rightPanelOpen,
+    schedulerRef,
+    selectedAppointment,
+    tasks,
+    tooltipPosition,
+    tooltipRef,
+    createAppointment,
+    deleteCurrentAppointment,
+    editCurrentAppointment,
+    onCurrentViewChange,
+    onAppointmentClick,
+    onAppointmentTooltipShowing,
+    onCellModified,
+    onCellClick,
+    updateAgenda,
+    setDate,
+    setSelectedAppointment,
+    toggleRightPanelOpen,
+  } = useSchedulerLogic();
+
+  const onTodayClick = () => {
+    setDate(new Date());
+  };
+
+  const resourcesList = useMemo(() => {
+    return calendarListItems
+      .reduce((res: CalendarListItem[], calendarList) => { return res.concat(calendarList.items); }, []);
+  }, [calendarListItems]);
+
+  const onSelectedCalendarsChange = useCallback((seletedCalendars) => {
+    const removedResourceFilters = seletedCalendars
+      .map((calendar) => calendar.id);
+
+    tasks?.filter((task) => {
+      return !removedResourceFilters.includes(task.calendarId);
+    });
+
+    tasks?.load().then(() => { updateAgenda(selectedAppointment?.data); });
+  }, [tasks, selectedAppointment]);
+
+  const onSelectedDateChange = useCallback((e) => {
```

Comment:
Looks like these methods (onSelectedDataChange, onSelectedCalendarsChange) should be moved to the hook as well.
Every function that is called inside is from the hook, so why move logic out of the hook?
In this case, setSelectedAppointment, updateAgenda and maybe setDate shouldn't be exported from the hook at all.

Reply:
The idea was that callbacks are used in calendar component and act like extensions to the scheduler logic
But if you think thats better to move all logic inside one hook and later split it i will agree